File: frontend/ksweb-app/src/app/pages/daily/daily.models.ts
export interface DailyListItem {
  dailyId: number;
  dailyNumero: number;
  dailyData: string;
  dailyUsuario: string;
}

export interface DailyListResponse {
  total: number;
  page: number;
  pageSize: number;
  totalPaginas: number;
  items: DailyListItem[];
}

export interface DailyFiltro {
  page: number;
  pageSize: number;
}

export interface DailyForm {
  dailyId: number | null;
  dailyNumero: number | null;
  dailyData: string;
  dailyUsuario: string;
}

export interface CriarDailyRequest {
  dailyUsuario: string;
}

export interface CriarDailyResponse {
  sucesso: boolean;
  mensagem: string;
  dailyId: number;
  dailyNumero: number;
}

export interface DailyRegistro {
  regId: number;
  dailyId?: number;
  regData: string;
  osId: number | null;
  regCliente: string;
  regDescricao: string;
  regStatus: string;
}

export interface DailyRegistrosResponse {
  dailyId: number;
  total: number;
  items: DailyRegistro[];
}

export interface DailyRegistroForm {
  regData: string;
  osId: number | null;
  regCliente: string;
  regDescricao: string;
  regStatus: string;
}

File: backend/KSWeb.Api/Models/DailyModels.cs
namespace KSWeb.Api.Models;

public sealed class DailyFiltro
{
    public int? DailyNumero { get; set; }
    public string? Usuario { get; set; }
    public DateTime? DataInicial { get; set; }
    public DateTime? DataFinal { get; set; }

    public int Page { get; set; } = 1;
    public int PageSize { get; set; } = 25;
}

public sealed class DailyResponse
{
    public int Total { get; set; }
    public int Page { get; set; }
    public int PageSize { get; set; }
    public int TotalPaginas { get; set; }
    public List<DailyItem> Items { get; set; } = [];
}

public sealed class DailyItem
{
    public int DailyId { get; set; }
    public int DailyNumero { get; set; }
    public DateTime DailyData { get; set; }
    public string DailyUsuario { get; set; } = string.Empty;
    public int TotalRegistros { get; set; }
}

public sealed class DailyNovaRequest
{
    public string DailyUsuario { get; set; } = string.Empty;
}

public sealed class DailyNovaResponse
{
    public bool Sucesso { get; set; }
    public string Mensagem { get; set; } = string.Empty;
    public int DailyId { get; set; }
    public int DailyNumero { get; set; }
}